### DISCUSSION TITLE
New setup for CommandStation-EX - Track power

### DISCUSSION THREAD

**Discussion by hiEddie** [DCC-EX/CommandStation-EX#251] [General]
I am a newbie on railway model. I just setup the Mega2560 + Motor shield (Deek-robot).  

After I turn on the track power, when I measure the track output from the Motor shield,  I got the result as around DC 0.02 to 0.05 V.  
Is it normal?  or something wrong with the setting?

Here is the status from serial monitor.

<* License GPLv3 fsf.org (c) dcc-ex.com *>
<* LCD0:DCC++ EX v4.0.0 *>
<* LCD1:Lic GPLv3 *>
<* ++ Wifi Setup Try 1 ++ *>
<* Wifi Check: [+IPD\0] *>
<* TIMEOUT after 200ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
<* TIMEOUT after 200ms *>
<* ++ Wifi Setup NO AT ++ *>
<* ++ Wifi Setup Try 2 ++ *>
<* Wifi Check: [+IPD\0] *>
<* TIMEOUT after 200ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
<* TIMEOUT after 200ms *>
<* ++ Wifi Setup NO AT ++ *>
<* ++ Wifi Setup Try 3 ++ *>
<* Wifi Check: [+IPD\0] *>
<* TIMEOUT after 200ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
AT\r\n\r\nOK\r\n<* Found in 1ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
ATE1\r\n\r\nOK\r\n<* Found in 4ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
AT+GMR
AT version:1.7.4.0(May 11 2020 19:13:04)
SDK version:3.0.4(9532ceb)
compile time:May 27 2020 10:12:22
Bin version(Wroom 02):1.7.4
OK
<* Found in 16ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
AT+CWJAP_CUR?\r\nNo AP\r\n\r\nOK\r\n<* Found in 6ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
AT+CWMODE_CUR=1\r\n\r\nOK\r\n<* Found in 5ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
AT+CWHOSTNAME="dccex"\r\n\r\nOK\r\n<* Found in 5ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
AT+CWJAP_CUR="KingKong","25475768"\r\nWIFI CONNECTED\r\nWIFI GOT IP\r\n\r\nOK\r\n<* Found in 6749ms *>
<* Wifi Check: [+CIFSR:STAIP\0] *>
AT+CIFSR
+CIFSR:STAIP<* Found in 3ms *>
<* Wifi Check: [0.0.0.0\0] *>
,"192.168.1.149"
+CIFSR:STAMAC,"a4:e5:7c:bc:d8:e2"

OK
<* TIMEOUT after 1000ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
AT+CIPSERVER=0\r\n\r\nERROR\r\n<* TIMEOUT after 1000ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
AT+CIPMUX=1\r\n\r\nOK\r\n<* Found in 3ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
AT+MDNS=1,"dccex","withrottle",2560\r\n\r\nOK\r\n<* Found in 7ms *>
<* Wifi Check: [\r\nOK\r\n\0] *>
AT+CIPSERVER=1,2560\r\n\r\nOK\r\n<* Found in 5ms *>
<* Wifi Check: [IP,"\0] *>
AT+CIFSR
+CIFSR:STAIP,"<* Found in 4ms *>
192.168.1.149"<* LCD4:192.168.1.149 *>
<* Wifi Check: [\r\nOK\r\n\0] *>

+CIFSR:STAMAC,"a4:e5:7c:bc:d8:e2"

OK
<* Found in 7ms *>
<* LCD5:PORT=2560 *>
<* Wifi Check: [\r\nOK\r\n\0] *>
ATE0\r\n\r\nOK\r\n<* Found in 4ms *>
<* ++ Wifi Setup CONNECTED ++ *>
<* MotorDriver currentPin=A1, senseOffset=0, rawCurrentTripValue(relative to offset)=668 *>
<* MotorDriver currentPin=A0, senseOffset=0, rawCurrentTripValue(relative to offset)=668 *>
<iDCC-EX V-4.0.0 / MEGA / STANDARD_MOTOR_SHIELD G-a26d988>
<* No I2C Devices found *>
<* MCP23017 I2C:x20 Device not detected *>
<* MCP23017 I2C:x21 Device not detected *>
<* Signal pin config: high accuracy waveform *>
<* LCD3:Ready *>
<* LCD2:Power Off *>
<p0>
PPA0
<* LCD3:Free RAM= 3126b *>

Also,  I can not get any status from the decoder through PROG track.

<* LCD0:Retry 17 2 Sum=5 *>
<* LCD0:Retry 17 1 Sum=6 *>
<r -1>
<* LCD2:Power On MAIN *>
<p1 MAIN>
PPA1
<* LCD2:Power Off *>
<p0>
PPA0

**Comment by FrightRisk** [comment]
You must measure track power with AC, not DC. That won't be accurate unless you have a "true RMS" meter,  but close enough to know if you have voltage. If all 4 lights are lit on the outputs after sending "<1>", then you have output and can just make sure it shows something reasonable on your meter.
To diagnose programing track issues, see this: https://dcc-ex.com/reference/software/diagnostic-d-ack-command.html

**Comment by hiEddie** [comment]
Thanks.
I use 14V DC input for Motor shield,  but I only get 5V as output after switching to AC meter checking.
Refer to document,  it should be around 14V AC by return.  Is it a littler bit low?

**Comment by hiEddie** [comment]
The decoder is Digitrax DZ123
Can not get response from PROG.

<* ACK baseline=5/14mA Threshold=25/74mA Duration between 4000us and 8500us *>
<* V0 cv=8 bit=7 *>
<* NO-ACK after 140mS max=6/17mA pulse=0uS samples=32 gaps=0 *>
<* V1 cv=8 bit=7 *>
<* NO-ACK after 145mS max=13/38mA pulse=0uS samples=76 gaps=0 *>
<* LCD0:Retry 8 2 Sum=7 *>
<* ACK baseline=3/8mA Threshold=23/68mA Duration between 4000us and 8500us *>
<* V0 cv=8 bit=7 *>
<* NO-ACK after 143mS max=6/17mA pulse=0uS samples=66 gaps=0 *>
<* V1 cv=8 bit=7 *>
<* NO-ACK after 145mS max=13/38mA pulse=0uS samples=76 gaps=0 *>
<* LCD0:Retry 8 1 Sum=8 *>
<* ACK baseline=3/8mA Threshold=23/68mA Duration between 4000us and 8500us *>
<* V0 cv=8 bit=7 *>
<* NO-ACK after 143mS max=6/17mA pulse=0uS samples=66 gaps=0 *>
<* V1 cv=8 bit=7 *>
<* NO-ACK after 144mS max=13/38mA pulse=0uS samples=76 gaps=0 *>
<* Callback(-1) *>
<r1|1|8 -1>

**Comment by Ash-4** [comment]
There is a consistent 21mA difference between V0 and V1 -- perhaps try this.  <D ACK LIMIT 15>
I don't see anything to suggest what the MIN should be.
The ACK defaults in the 4.0.0 version: LIMIT 60, MIN 4000, MAX 8500.

Keep in mind that these small readings are not normally seen.  Check voltage at the motor board and across the tracks to see that it is consistent -- checking for wiring connections.  Clean tracks.  Clean wheels.  Re-seat the motor board in the Mega.  Leave off the wifi shield until you are able to read CVs on the programming track.

What is your DC power supply 14V ?   Looks good, but what is the current rating?  Also check polarity.  It doesn't have to be a large power supply to read CVs; I've had success using a 12V 2A wall wart.

Are you supplying power to the Mega using the barrel connector with 7+ volt power? 9V is often recommended for this; below 7V is a problem.  While you may be able to load the sketch using the USB for power, don't assume USB is providing adequate power with wifi shield or other accessories connected.

**Comment by jlouage** [comment]
I have the same problem. I am using arduino Mega with Arduino Motor Shield R3. I am testing the voltage on the track I'm getting 5.72 AC if my power supply is 12V DC. I increased the voltage to 19V to teach the 10v AC. This i think will fry my arduino Motor Shield because the maximum is 12V. If i put my train on program track. The train will make a noise. If i send the <R> command the train will make sound like the old modem when calling a number but it gives me error can't get the address. I tried 2 different train with the same issue. Any help?

**Comment by habazut** [comment]
> I have the same problem. I am using arduino Mega with Arduino Motor Shield R3. I am testing the voltage on the track I'm getting 5.72 AC if my power supply is 12V DC.

Do you measure with a meter that is certified for high frequency AC? Probably not.

Are both motor shield LED of the output you want to use on?
If you measure DC betwee output screw and GND, do you get approx 6V from all output terminals that are on?

> I increased the voltage to 19V to teach the 10v AC. This i think will fry my arduino Motor Shield because the maximum is 12V.

Probably not but that's the wrong approach.

>  If i put my train on program track. The train will make a noise.

That sounds like no decoder in train. What decoder model do you think you have in the loco and can you send a picture in our Discord?

Harald.

**Reply by jlouage** [reply]
Thanks habazut for your reply.

I have attached photo for the multimeter values with the power supply and the motor shiled when it is on ![image](https://github.com/DCC-EX/CommandStation-EX/assets/96912852/2cb607ce-59e4-4939-9255-9b863e9d5003).
You can see the leds in the motor shields are on, the value of the multimeter is around 6.5v (same values if I put the multimeter directly on the screws of the motor shields AC and the amps on the power aupply is 0.017A.

Find the video when I put the Loco on the track and turn it on. And when I send the <R> command tue sound and it gives me error.

https://drive.google.com/file/d/13c10ZSiLQ2nryztoJR_Z7KwPhxLCl0Mj

The loco is the Santa Fe from Bachmann with a decoder NMRA 8-pin plug.
by![image](https://github.com/DCC-EX/CommandStation-EX/assets/96912852/a4ddaf04-02e3-4d6d-a1f4-2cff32f07c83)

**Comment by Ash-4** [comment]
I prefer to measure track voltage on DC scale.  Each rail vs GND will be about 50% of track power supply -- a 50% PWM.

Your loco box says DCC Ready.  It is ready to have a DCC decoder installed.
Without a decoder, it will run in DC mode.  Assign a loco ID to the track and drive the DC train with a throttle.
https://dcc-ex.com/trackmanager/index.html#trackmanager-commands

**Comment by cameroncooke** [comment]
I know this is a very old discussion but I have the exact same problem with track DC voltage, I'm getting roughly half the input voltage to the deek-robot motor shield. I'm also using a bench power supply and a Fluke 117 Multimeter to measure DC voltage. If I input 15v AC to the motor shield, I get about 9v DC on track or when testing directly on the motor shield. Has anyone got to the bottom of this?

**Comment by Asbelos** [comment]
Motor shield input should be DC !

**Comment by Asbelos** [comment]
Motor shield input must be DC

**Comment by habazut** [comment]
Is your multimeter "true RMS" for a variable frequency recangle wave? 
Probably not.
Harald.

**Comment by cameroncooke** [comment]
Okay I’m not entirely sure what that means but I’ve now used a small
oscilloscope and can see it measures rms voltage at the correct voltage.

I’m learning about electronics so please go easy on people like me. We all
have to start somewhere.

Cameron.

**Reply by FrightRisk** [reply]
Hope to see you on Discord. The entire team and many of our users can help you along. We are happy to educate you on all sorts of subjects. Even a scope does not show what a bi-polar signal is, but it at least lets you measure it accurately ;) To a scope, it looks like a square wave. But the actual DCC signal never goes negative, it is just always positive with reference to the other rail. It has no "ground". A scope will show a square wave with a peak-to-peak value of say 24V, but in reality the voltage never goes above12. So reading either half of the wave will give you what you are looking for.

A "True RMS" meter is designed to measure a 60Hz sine wave accurately. It tells you the the heating value of a waveform that would equal the equivalent DC voltage. Higher frequencies and non-sinusoidal waveforms (like the DCC signal) will confuse a standard averaging meter and any RMS meter that isn't "true rms" and has the smarts in it to recognize the wave and compute the correct value. Without that meter, your AC readings could be 10 to 40% off in either direction, high or low.

Measuring DC from each rail to a GND pin (like the sleeve of your input power supply) will give you a good idea what you have without a scope. Since the meter averages the voltage, if the signal is "on" relative to ground half the time, you will read about half the voltage. If you measure the MAIN track and then measure the PROG track, you will see a slight difference because the packets being sent, and the average time the rail is at full voltage, is different. But both are close to have the input voltage minus the drop in the motor shield.

**Comment by Ash-4** [comment]
You may find it helpful to use the discord site, where we can have an interactive discussion, post pictures, etc.  And you will likely see a reply within a few minutes.  Open a ticket, or use an existing topic, such as # motor-shields.
https://dcc-ex.com/support/index.html#preferred-option-discord

When measuring voltage on the track outputs, each rail vs GND is similar to a 50% PWM and can be measured on the DC scale.  (And that is also how I connect a scope to see the voltage.)

**Comment by cameroncooke** [comment]
That’s all makes sense thank you. So it would seem the voltage is correct
it’s just my ability to measure which is flawed. I’m pretty confident the
voltage is correct. Thank you so much. I will for sure join the discord
channel.

Cameron.
